### PR TITLE
Infra/496/add prettier

### PR DESCRIPTION
**Pull request by joshuacrass**
Added prettier and pre-commit hooks for linting and prettier (#496) 

The checks will only run on staged files. This will only "check" the files and not fix them. The commit will fail if any staged files do not pass.  

If you need/want to fix them, you will need to run `yarn run prettier --write <filename>`.

There are a few options to prettier your files. I would recommend, if possible, to add the prettier plugin to your editor.

Note that as a result of editing eslintrc and package.json, there are several whitespace changes. We must also keep in mind that editing any file that does not currently conform to prettier standards will also be affected by several formatting changes.

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1 @@
+package-lock.json
```
Comment (ryan-roemer) [inline .prettierignore]: Let's also remove `package.json`. `yarn` and `npm` write these in incompatible ways IIRC.

Comment (joshuacrass) [reply .prettierignore]: ✅

**Comment by sxzhao** [comment]
@joshuacrass this looks good! Please fix the merge conflicts. If all looks good I’ll go ahead and merge.

**Comment by zhaostan** [comment]
@joshuacrass thanks for fixing the merge conflicts! Looks like now CI is failing, if you sync with master it should fix it. Thanks!

**Pull request merged**